api/controllers/FileUploadController.js

```diff
@@ -93,6 +93,17 @@ const FileUploadController = {
         }
     },
 
+    _maxFileLimitCheck(req) {
+        const totalFilesUploaded = req.session.eApp.uploadedFileData.length;
+
+        if (totalFilesUploaded > maxFileLimit) {
+            req.flash('genericErrors', [
```

Comment:
is the user prevented from proceeding with the form if genericErrors is populated and if so, where?

Reply:
Yes the user is prevented from proceeding, there is [some code in the markup](https://github.com/UKForeignOffice/loi-application-service/blob/develop/views/eApostilles/uploadFiles.ejs#L184) to hide the 'Continue' button on the upload documents page until the user removes the files they don't want uploaded.

Reply:
Thanks.  Correct me if I'm wrong but this coupled with the check in EAppReferenceController this covers the UI (disabling the button) and a bot/hack submitting the form data to the /additional-reference endpoint to attempt to progress the application without having uploaded a valid number of docs, right?

Reply:
yes that's correct